### PR TITLE
Feature: mentor card

### LINKED ISSUE #103: Mentor Profile card

### PR DESCRIPTION
**Pull request by rpollock03**
Resolves #103 (and part of 113) 

Add **reusable mentor card component**. Also add the **parent component to search for mentors**, including retrieval of mentors from Firebase. **Components displayed static side-by-side for now, but will be displayed in carousel as part of future pull request** (@JakubFrackowiak has built a universal carousel/slider component that has not yet been merged).

Screenshot 2022-11-11 at 09 00 36

**Important**: There was a decision about whether to have separate 'mentors' collection, or to include 'mentor' data as part of hero data. After discussing with Oli I have gone with the second option which is more efficient, because we don't have to make two separate requests for each mentor (considering shared name, level, bio, profile picture etc). 

So I have updated the 'hero' type to include mentor data like below. There will likely be more properties to add to this, but this is what is needed for now. The 'isMentor' boolean allows us to filter for active mentors, without user having to delete the mentor object each time which is the best way to do this I think.

Screenshot 2022-11-11 at 09 11 46

On Firebase:

Screenshot 2022-11-10 at 20 03 43

### CODE REVIEW

File: components/SearchMentors/MentorCard.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+export function MentorCard({ mentor }: MentorProps) {
+  const MentorPicture = styled(StorageImage)({
+    objectFit: "cover",
+    height: 290,
+  })
+
+  return (
+    <Box sx={{ maxWidth: 280 }}>
+      <Stack spacing={3}>
+        <MentorPicture storagePath={`${mentor?.profilePicture}`} />
```
Comment (piotrostr) [inline components/SearchMentors/MentorCard.tsx:20]: null chaining is redundant, this is checked in the parent component

File: components/SearchMentors/MentorCard.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+          <Typography
+            variant="body1"
+            fontWeight={200}
+            color="text.secondary"
+          >{`${mentor?.name.first} ${mentor?.name.last}`}</Typography>
+          <Typography
+            variant="body1"
+            fontWeight={400}
+            color="primary.main"
+          >{`Between £${mentor?.mentor.minRate} - £${mentor?.mentor.maxRate} p/h`}</Typography>
```
Comment (piotrostr) [inline components/SearchMentors/MentorCard.tsx:34]: style error, not formatted correctly

File: components/SearchMentors/MentorCard.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import LinesElipsis from "react-lines-ellipsis"
+import styled from "styled-components"
+import { StorageImage } from "reactfire"
+import { Hero } from "storage/hero"
+
+interface MentorProps {
+  mentor: Hero
+}
+
+export function MentorCard({ mentor }: MentorProps) {
```
Comment (piotrostr) [inline components/SearchMentors/MentorCard.tsx:11]: add some slight padding, maybe justify the text a bit? the screenshot looks a bit wonky

File: components/SearchMentors/index.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  const { status: mentorsStatus, data: mentors } =
+    useFirestoreCollectionData(mentorsQuery)
+
+  return (
+    <Stack spacing={4} alignItems="center" sx={{ overflow: "clip" }}>
+      <Container>
+        <Stack spacing={6}>
+          <Stack direction="row">
+            {/*TO DO: ADD REUSABLE SLIDER COMPONENT */}
+
```
Comment (piotrostr) [inline components/SearchMentors/index.tsx:23]: why empty line?

**Review by piotrostr — CHANGES_REQUESTED** [review]

File: components/SearchMentors/MentorCard.tsx (hunk 3 above)
Comment (rpollock03) [reply components/SearchMentors/MentorCard.tsx:11]: Done. Apologies, was planning to finalise styling when adding to the universal slider.

**Comment by rpollock03** [comment]
Hi @piotrostr I fixed styling issue and other nits. Should be good to merge. I will update this to use mentor cards with the new universal slider component when it is built/merged. Thanks. 
Screenshot 2022-11-15 at 19 14 37

**Review by piotrostr — APPROVED** [review]

**Comment by piotrostr** [comment]
looking good, pls merge @rpollock03

**Pull request merged**